src/rviz/default_plugin/path_display.cpp

```diff
@@ -80,35 +101,108 @@ void PathDisplay::reset()
   updateBufferLength();
 }
 
+
+void PathDisplay::updateStyle()
+{
+  LineStyle style = (LineStyle) style_property_->getOptionInt();
+
+  switch( style )
+  {
+  case LINES:
+  default:
+    line_width_property_->hide();
+    break;
+
+  case BILLBOARDS:
+    line_width_property_->show();
+    break;
+  }
+
+  updateBufferLength();
+}
+
+void PathDisplay::updateLineWidth()
+{
+  LineStyle style = (LineStyle) style_property_->getOptionInt();
+  float line_width = line_width_property_->getFloat();
+
+  if(style == BILLBOARDS) {
+    for( size_t i = 0; i < billboard_lines_.size(); i++ )
+    {
+      rviz::BillboardLine* billboard_line = billboard_lines_[ i ];
+      if( billboard_line ) billboard_line->setLineWidth( line_width );
+    }
+  }
+  context_->queueRender();
+}
+
+void PathDisplay::updateOffset()
+{
+  scene_node_->setPosition( offset_property_->getVector() );
+  context_->queueRender();
+}
+
 void PathDisplay::destroyObjects()
 {
+  // Destroy all simple lines, if any
   for( size_t i = 0; i < manual_objects_.size(); i++ )
   {
-    Ogre::ManualObject* manual_object = manual_objects_[ i ];
+    Ogre::ManualObject*& manual_object = manual_objects_[ i ];
     if( manual_object )
     {
       manual_object->clear();
       scene_manager_->destroyManualObject( manual_object );
+      manual_object = NULL; // ensure it doesn't get destroyed again
```

Comment:
My guess is so that you can do this and set the pointer in the vector to `NULL`?